src/pages/Search.vue

```diff
@@ -181,10 +192,56 @@ export default class Search extends Vue {
         this.gradeFilter,
       );
     }
+    this.hasFilteredResults = true;
 
     this.setSearchResults(buildingsResults);
   }
 
+  /** Handles clicks on numeric columns in
+   * BuildingsTable and sorts (desc, asc) */
+  handleSort(field: string): void {
+    // if clicking on the same field,
+    // this will toggle between 'desc' and 'asc'
+    if (this.sortedField === field) {
+      this.sortedDirection = this.sortedDirection === 'desc' ? 'asc' : 'desc';
+    } else {
+      // else if clicking on a new field, set sortedField state
+      // to the new field AND set initial sortedDirection to 'desc'
+      this.sortedField = field;
+      this.sortedDirection = 'desc';
+    }
+
+    /** If there are filtered results (type, grade, search results)
+     * pass to runSort. If not, sort all. */
+    let buildingsToSort;
+    if (this.hasFilteredResults) {
+      buildingsToSort = [...this.searchResults];
+    } else {
+      buildingsToSort = [...this.$static.allBuilding.edges];
+    }
+
+    this.runSort(buildingsToSort);
+  }
+
+  /** Called from handleSort, this function sorts
+   * according to sortedField and sortedDirection state values */
+  runSort(buildings: any): void {
```

Comment:
I think buildings _should_ have the type `Array<IBuildingNode>`, which will then make other stuff easier